byoqm/source_repository/source_repository.py

```diff
@@ -128,6 +142,29 @@ def _inspect_file(self, file_path: Path) -> FileInfo:
 
         return FileInfo(file_path, encoding, programming_language)
 
+    def _get_ignored_files(self):
+        ignored_files = self._get_ignored_paths()
+        ignored_files.extend(self.get_ignored_file_extensions())
+        return ignored_files
+
+    def _get_ignored_paths(self):
+        with _IGNORE_FILE_PATH.open("r") as file:
+            ignored_files = []
+            for line in file:
+                path = Path(line.rstrip())
+                if path.is_dir():
+                    ignored_files.extend(path.glob("*"))
+                if path.is_file():
+                    ignored_files.append(path)
+        return ignored_files
+
+    def get_ignored_file_extensions(self):
+        with _IGNORE_FILE_EXTENSIONS_PATH.open("r") as file:
+            ignored_files = []
+            for extension in file:
+                ignored_files.extend(self.src_root.glob(f"**/*{extension.rstrip()}"))
+        return ignored_files
```

Comment:
Git views each line in the `.gitignore` file as a file pattern that can be matched on. We could do the same, and then use [fnmatch.filter](https://docs.python.org/3/library/fnmatch.html#fnmatch.filter) to match on these patterns  (or not match since we want to ignore these patterns). This should allow us to use the same file on everything as we just view each line of the ignore file as a regex essentially.